New option: Wrap titles

math-GH opened the pull request:
Closes #4428
Closes #3740
Ref. #3997

Changes proposed in this pull request:

- add an option to wrap long titles in normal view and global view
- options for big screens ("Desktop") and small screens ("mobile devices")

![grafik](https://user-images.githubusercontent.com/1645099/188701673-b321b195-3fe4-42f1-b147-7ae6e9e3949b.png)


How to test the feature manually:

1. go to `configuration` -> `reading`
2. enable/disable the checkboxes
3. check the titles in normal view or global view

Pull request checklist:

- [x] clear commit messages
- [x] code manually tested
- [ ] i18n

math-GH committed:
Configuration

math-GH committed:
improve class position

math-GH committed:
move .horizontal-list .item from themes to template.css

math-GH committed:
wip

math-GH committed:
fix

math-GH committed:
RTL

math-GH commented:
Right now the text is still hard coded.

I am waiting for your feedback first.

math-GH committed:
Merge branch 'edge' into wrap-titles

Frenzie commented on p/themes/base-theme/template.css:
```diff
@@ -1066,18 +1071,19 @@ input[type="search"] {
 }
 
-.flux .item.website {
+.flux .flux_header .item.website {
```
As an aside, the PR currently has this visual regression:
<img src=https://user-images.githubusercontent.com/202757/188716141-f0707d75-5c0e-4daf-ba18-f1b597671213.png width=40%>

math-GH replied:
thanks! fixed

Frenzie commented:
I'm not in favor of making this an option, or at least not in this form. Something similar can be done with the number of lines fairly easily along these lines, defaulting to 3 or 4.

```css
.flux .item.title a {
 display: -webkit-box;
 -webkit-line-clamp: 3;
 -webkit-box-orient: vertical;
}
```

Then you can override the `-webkit-line-clamp` attribute, whether with a CSS variable or more traditionally.

It'll then simply grow up to the desired number of lines before being cut off by ellipses. These properties are also supported by Firefox.

math-GH committed:
fix mobile view wide feed name column

math-GH committed:
improved: choose the numbers of line 1-4

math-GH commented:
Now it is possible to chose how many lines maximal should be displayed (1 till 4)
![grafik](https://user-images.githubusercontent.com/1645099/189435787-dbf66695-9e6f-44da-9893-13c65147d7dc.png)

math-GH committed:
Merge branch 'edge' into wrap-titles

Frenzie commented:
Nice! :+1: 

But I'd like to set it to something a lot higher, preferably just a regular `input type=number`. :-) 4 is a good default maximum but for example a max of about 12 would be good for preventing titles extending much past half a phone screen just in case. That's de facto no ellipsis rather than truly no ellipsis. That being said, I'm not sure I've ever actually seen more than 4.

Frenzie commented on app/views/configure/reading.phtml:
```diff
@@ -68,6 +68,27 @@
 			</div>
 
+			<div class="form-group">
+				<label class="group-name" for="wrapTitle">Wrap long titles in Normal View and Global view</label>
```
Something like clip or cut off, not wrap.

Frenzie replied:
Hm, although on second thought it'll wrap too… but wrap is a bit technical.

Use multiple lines for… ?

math-GH replied:
Would it be better just to have `titles in Normal View and Global view`?

Then we need a good word for "no wrap" as first select list item

Frenzie replied:
I'd only looked at the CSS in detail; I misinterpreted "no wrapping" to mean "no clipping". One line doesn't make a lick of sense to me, hence the misunderstanding.

math-GH replied:
I am not sure if I understand you correct.
Could you please give your idea of the left text (`wrap long titles in Normal View and Global view`) and of the first select option (`no wrap`)?

math-GH commented:
> But I'd like to set it to something a lot higher, preferably just a regular `input type=number`. :-) 4 is a good default maximum but for example a max of about 12 would be good for preventing titles extending much past half a phone screen just in case. That's de facto no ellipsis rather than truly no ellipsis. That being said, I'm not sure I've ever actually seen more than 4.

Let's wait here for a feature request in future. It is still a title, not the whole article ;)

Frenzie commented:
Yes, that's my feature request so I can stop using my custom CSS rather than having to keep it or replace it with another. ;-)

Here's an extremely rare example of more than 4 lines of title. I don't find it bothersome. Perhaps I would if it were more common. I used to run a max-height: 40vh on it just in case, but I don't think it's ever been reached by anything real.

### Laptop
<img src=https://user-images.githubusercontent.com/202757/189452250-0903de15-be05-4c99-81e5-1fd705e774fe.png width=40%>

### Mobile
<img src=https://user-images.githubusercontent.com/202757/189452520-93f4e21e-b2c5-4e49-9f50-15c39da5720e.jpg width=40%>

### Artificial gigantic title cut off at 40vh (in this case ~12 lines)
<img src=https://user-images.githubusercontent.com/202757/189453373-1d477954-1939-4efc-b4da-36c03bc0862e.png width=40%>

math-GH commented:
I understand. That is weird. Anything that has a user input is impossible (?) to handle it with CSS. For each number I have to prepare a CSS class, because of  `-webkit-line-clamp: 3;`

I could add:
- 5 lines
- unlimited (show the full title). Here you can do you individual height cut off with Custom CSS, but you do not need to manage all the other stuff in Custom CSS anymore (hover with the mouse over the line, vertical alignment of the icons etc.). And you can configure it individually for mobile and desktop view.

Frenzie commented:
D'oh, my bad, I skimmed too quickly. I assumed things and when you assume…

Anyway, I figured you'd be using `Minz_View::appendStyle` to override it on the page. But you can actually also do what you were doing here.

```css
.flux .item.title a {
 --clamp: 3;
 -webkit-line-clamp: var(--clamp);
}
```

And then on the element to override:
```php
style="--clamp: <?php FreshRSS_Context::$user_conf->clamp_title_mobile ?>"
```

math-GH committed:
Merge branch 'edge' into wrap-titles

math-GH committed:
Merge branch 'edge' into wrap-titles

math-GH commented:
> And then on the element to override:
> 
> ```
> style="--clamp: <?php FreshRSS_Context::$user_conf->clamp_title_mobile ?>"
> ```

inline style is not allowed because of the Content Security Policy

Frenzie commented:
You expressed a preference for doing it that particular way. As I said I think it's much more logical to me to simply use PHP to set something like:

```css
.flux  .item.title a {
	-webkit-line-clamp: $clamp_value;
}
```

math-GH committed:
Improved: Alternative Dark theme (#4635)

* background colors

* border

* Update adark.css

* font-color

* empty feed color

* slider

* color input improved

* menu nav hover

* nav: highlight font color while hovering

* lighter color btn icon on hover

* btn-attention background color

* empty feed color

* delete .log CSS

* delete white background .icon when feed active

* article header+footer

* bigMarkAsRead button

* .content hr

* dropdown triangle color

* rename variables

* notification colors

* border colors

* last left colors

* reorder

* hover flux icons

* config dropdown menu mobile view shadow+border

* dropdown menu hover color

* close aside hover

* slider-shadow

* icon hover

* toggle aside

* improved contrast blue

* contrast border color hover

* kbd styled

* hover on form elements highlight font color

* font colors instead background color

* link in alert

* contrast attention font color

* fix stream last item border

* font color of author in stream

* lighter color for headlines in article

* nav: active menu entry in lighter font color

* Update adark.rtl.css

* fixes

* fix favicon hover

* fix

math-GH commented:
> You expressed a preference for doing it that particular way. As I said I think it's much more logical to me to simply use PHP to set something like:
> 
> ```css
> .flux  .item.title a {
> 	-webkit-line-clamp: $clamp_value;
> }
> ```

I have no idea how to transfer the value from PHP to CSS, when I cannot use inline CSS

math-GH committed:
Update main.js (#4652)

math-GH committed:
Origine.css (#4653)

math-GH committed:
Rename template.css (#4644)

* Fix conflicts

* Update doc

* Misses

* Update prefix in CSS

Co-authored-by: Alexandre Alapetite <alexandre@alapetite.fr>

Frenzie commented:
I can probably look into it next week if you like, but I think at least one potential answer is self-evident from the question. Put another way, see CustomCSS.

math-GH committed:
all themes (#4663)

Commit:
Update Japanese translation (#4668)

* Update Japanese translation

* fix translations syntax

math-GH committed:
Improved: CSS .toggle_aside (#4664)

* all themes

* fixed order

* fix end of file

* fixed swage

SoldierCorp committed:
UPDATE: Spanish translations [i18n] #4656 (#4676)

NicolasPereira committed:
[Feature - Hacktoberfest] - Add Admin transalation pt br  (#4669)

* feat (translate): translate Registration form

* feat(translate): add translaction notform term

* feat(translate): add translate nolimit term

* feat(translate): add translate ptBR setaccountsnumber term

* feat(translate): add translate ptBR disabled term

* fix(translate): fix disabled translate ptBR term

* feat(translate): add translate ptBR enabled term

* feat(translate): add translate ptBR title term

* feat(translate): remove //todo comments

* feat(translate): add translate ptbr level term

* feat(translate): add translate ptbr message term

* feat(translate): add translate ptBR timestaamp term

* feat(translate): add translate ptBR term

* feat(translate): add translate ptBR both term

* feat(translate): add translate ptBR header term

* fix: remove TODO

* feat(translate): add translate ptBR header term

* feat(translate): add translate ptBR none term

* feat(translate): add translate ptBR above_title term

* feat(translate): add translate ptBR none term

* feat(translate): add transaction ptBR with_authors term

* feat(translate): add transaction ptBR feed_title term

* feat(translate): add translate ptBR term

* fix(translate): fix translate

* feat(translate): add translate ptBR both term

* feat(translate): add translate ptBR tags -> footer term

* feat(translate): add translate ptBR tags->header term

* feat(translate): add translate ptBR tags->none term

* feat(translate): add translate ptBR tags_max term

* feat(translate): add translate ptBR help term

* feat(translate): add translate ptBR headline

* feat(translate): add upong_gone term translate ptBR

* feat(translate): add deprecated terms translatePTBR

* feat(translate): add ptBR translate in gen.php

* feat(translate): add translate ptBR for sub.php

* Tab instead of white space

* another fix

Co-authored-by: mathContao <math-home@web.de>

SoldierCorp committed:
Update CREDITS.md (#4678)

math-GH committed:
Improved: Subscription Management: Show the position number (#4679)

* the code

* improvement

* fix doubled parameter

aledeg committed:
Allow redirection after login (#4654)

Before, if you've tried to reach a page without being logged, you'll be
automatically redirected to the index page after login.
Now, the original page is used after login.

Fix #3663

math-GH committed:
Improved: Frss.css - font size + line height with 'rem' values (#4671)

* headlines h1-h3

* delete for <sup> trust the browser

* .icons = 1rem

* line-height WIP

* .flux .item

* paragraphs

* kbd

* button.as-link

* .form-advanced-title

* set line-height 1.5 as default

* .flux .item.title .summary

* .flux .item.title .author

* .content > header, .content > footer

* popup

* last lines

* RTL CSS

* imrpove jumpy line

* Origine Compact

* Swage

* FRSS CSS RTL

* deleted: br

* deleted: font-size: 16px

* RTL CSS

aledeg committed:
Add documentation to work with feed snapshots (#4598)

math-GH committed:
Improved: Article summary. Cut the text with better CSS (#4666)

* phtml + css

* trim the white spaces that comes from stripping

math-GH committed:
Improved: better semantic HTML for boxes (#4665)

* add h1 title on views

* add h2 titles into box-title

* thin scrollbar in box-content

math-GH committed:
Improved: Remove <li> as separator (#4597)

* templates

* themes

* fix

* fix

* fixes

* fix

NicolasPereira committed:
feat: add new icon (#4673)

math-GH committed:
fix (#4685)

math-GH committed:
All (S)CSS files (#4690)

Alkarex committed:
Compat PHP 8.1 strip_tags (#4688)

#fix https://github.com/FreshRSS/FreshRSS/issues/4687

Cyb10101 committed:
Slider: Block Page for one scrollbar; Scrollbar width normal (#4692)

* Block page if slider is open; Hides the double page scrollbar

* Slider scrollbar default width

* Update frss.rtl.css

* Add user Cyb10101 to credits

* make fix-all

* make fix-all (second pass)

Co-authored-by: mathContao <math-home@web.de>
Co-authored-by: Alexandre Alapetite <alexandre@alapetite.fr>

math-GH committed:
fix (#4697)

math-GH committed:
fix (#4696)

Commit:
Updated Czech (CZ) 🇨🇿 [i18n] Translation (#4670)

* Update admin.php

Update missing translation on missing Czech strings

* Update conf.php

Updated missing Czech translation
Fixed typo on one row

* Update gen.php

Added missing Czech strings

* Update sub.php

Added missing Czech strings

* Update sub.php

* Update conf.php

Fix missing Czech translations

* Update admin.php

Decided to not use abbreviation

* Update sub.php

Update more missing czech translations

* Update CREDITS.md

Adding myself to Credits list

* Update sub.php

Fix translation

* Update sub.php

Updated Czech translate

Co-authored-by: mathContao <math-home@web.de>

Pyr0x1 committed:
Implemented italian translation (#4694)

* Translated admin.php

* Translated conf.php

* Translated feedback.php

* Translated gen.php

* Translated index.php

* Translated install.php

* Translated sub.php

* Translated user.php

* make fix-all

* fixed apostrophe

* fixed missing TODOs in sub.php

* A few more details

Co-authored-by: Alexandre Alapetite <alexandre@alapetite.fr>

Alkarex committed:
Fix regression 4688 (#4700)

Minor regression from https://github.com/FreshRSS/FreshRSS/pull/4688

Alkarex committed:
Fix OPML import special (#4708)

#fix https://github.com/FreshRSS/FreshRSS/issues/4704
From https://github.com/FreshRSS/FreshRSS/pull/4407

Alkarex committed:
Add support for custom XPath date/time format (#4703)

* Add support for custom XPath date/time format
#fix https://github.com/FreshRSS/FreshRSS/issues/4701
Improvement of https://github.com/FreshRSS/FreshRSS/pull/4220

* Format is not XPath

* Remove TODOs in en-GB

batchennings committed:
Main font color from blue to regular dark (#4709)

* Main font color from blue to regular dark

* newline added at the end of rtl css file

Co-authored-by: patjennings <patjennings@MacBook-Pro-de-patjennings.local>

Alkarex committed:
Fix encoding CSS filter (#4707)

#fix https://github.com/FreshRSS/FreshRSS/issues/4705

CilgaIscan committed:
Complete Turkish translation (#4715)

* Complete Turkish translation

* make fix-all

Co-authored-by: Alexandre Alapetite <alexandre@alapetite.fr>

CilgaIscan committed:
Update CREDITS.md (#4716)

Add Çılga İşcan Tercanlı to contributors list.

Alkarex committed:
Update GitHub Actions version (#4717)

https://github.blog/changelog/2022-09-22-github-actions-all-actions-will-begin-running-on-node16-instead-of-node12/

Alkarex committed:
i18n typographic quotes (#4714)

* i18n typographic quotes

* Missing whitespace

* Update app/i18n/nl/conf.php

Co-authored-by: Frans de Jonge <fransdejonge@gmail.com>

* Fix German-style typographic quotes

* Fix missing closing ”

* Fix Dutch quotes

* French typography

* <em>lazy load</em>

Co-authored-by: Frans de Jonge <fransdejonge@gmail.com>

gulyapulya committed:
Translate to Russian completed (#4719)

* Translate to Russian completed

* make fix-all

Co-authored-by: Alexandre Alapetite <alexandre@alapetite.fr>

nextdoorpanda committed:
adds greek translation (#4718)

* adds greek translation

* update translation

* make fix-all

* Typographic quotes merge

Co-authored-by: Alexandre Alapetite <alexandre@alapetite.fr>

amrulizwan committed:
Add id-🇮🇩 Translation (#4706)

* Add id-🇮🇩 Translation

* Update id Translation for admin.php

* make fix-all

* Bahasa Indonesia

* Fix typos config

* Update user.php

* make fix-all

* Typographic quotes

* make fix-all

Co-authored-by: Alexandre Alapetite <alexandre@alapetite.fr>

acbgbca committed:
Added Linkding as a sharing method #4720 (#4721)

* Added Linkding as a sharing method #4720

* Fixed typo in docs #4720

Alkarex committed:
make fix-all

nextdoorpanda committed:
Update CREDITS.md (#4722)

Co-authored-by: Alexandre Alapetite <alexandre@alapetite.fr>

gulyapulya committed:
Added myself to CREDITS.md (#4724)

* Added myself to CREDITS.md

* Update CREDITS.md

SoldierCorp committed:
UPDATE: Spanish translations 🇪🇸🇲🇽 [i18n] (#4725)

* UPDATE: Spanish translations [i18n] #4656

* ADD: Spanish translations for conf, get and sub files.

* FIX: Typo in comment

aledeg committed:
Change new article banner size (#4735)

Only on origin-compact theme

aledeg committed:
Change default values to have predictable default state (#4736)

math-GH committed:
fix: jumping lines (#4699)

* fix

* fix Origine compact

* fix Ansum

* fix Mapco

Co-authored-by: Alexandre Alapetite <alexandre@alapetite.fr>

jangernert committed:
Update NewsFlash app entry (#4650)

* Update NewsFlash app entry

* update fr readme & remove FeedReader

* Tested

Co-authored-by: Alexandre Alapetite <alexandre@alapetite.fr>

acbgbca committed:
Updated credits for PR 4721 (#4739)

aledeg committed:
Remove ajax parameter when modifying the request (#4743)

Before, when clicking on the author name to use as a filter, the displayed page
was a result page from the ajax query. This was due to the use of the ajax
parameter in the query.
Now, the ajax parameter is removed from the query before displaying the filters.

Fix #4726

aledeg committed:
Fix i18n tool when adding parent translation (#4742)

See #4737

math-GH committed:
Improved: Origine theme (colors in vars) (#4693)

* wip

* wip

* active btn = darker icon

* wip

* flux font-size

* wip

* mouse hover feed aside bar

* wip

* wip

* wip

* Update origine.css

* wip

* fix

* font-color

* fixes

* wip

* alerts

* alert: more contrast font colors

* notifications

* notification improved font colors

* wip

* wip

* text-shadow

* footer-border

* feed tree: hover icon

* box-shadow

* box-shadow + rename font-color-666

* wip

* wip

* fixes

* header-height

* RTL CSS

* dropdown background color

* improved mobile header

* wip

* fix

* Update frss.rtl.css

* active buttons highlighted

Alkarex committed:
Fix i18n el: Ελληνικά (#4754)

tnt committed:
Fix broken link to the updating instructions (#4753)

aledeg committed:
Add tests on i18n classes (#4756)

Alkarex committed:
Fix curlopt options for HTML+XPath (#4759)

Improvement of https://github.com/FreshRSS/FreshRSS/pull/4220

Alkarex committed:
GReader API doc and improvements (#4763)

* GReader API doc and improvements
#fix https://github.com/FreshRSS/FreshRSS/issues/4762

* Minor slash

* HTTPS for web.archive.org

math-GH committed:
Fix 4712 (#4750)

* fix

* fix

* fix nord theme (a bit)

* fix

math-GH committed:
fix 4744 (#4751)

* fix

* fix

Alwaysin committed:
Correct manage link on stats page (#4768)

Wasn't working before

acbgbca committed:
Added bottom nav padding for iOS #4740 (#4741)

* Added bottom nav padding for iOS #4740

* Updated RTL css #4740

Co-authored-by: maTh <math-home@web.de>

math-GH committed:
Fixed: "tag" -> "label" (#4770)

* tag -> label (all english texts)

* tag -> label (German)

* marked as DIRTY

* Update app/i18n/nl/feedback.php

Co-authored-by: Frans de Jonge <fransdejonge@gmail.com>

* Update app/i18n/nl/conf.php

Co-authored-by: Frans de Jonge <fransdejonge@gmail.com>

Co-authored-by: Alexandre Alapetite <alexandre@alapetite.fr>
Co-authored-by: Frans de Jonge <fransdejonge@gmail.com>

math-GH committed:
fix (#4775)

aledeg committed:
Fix integration page (#4774)

Before, when adding a simple integration, the delete button was missing until
the configuration was saved. This behavior was introduced in #4269, but I think
that it was a regression.
Now, the delete button is back and everything is working as it should.

In the process, I've extracted templates in attributes and add them in template
tags. This way, it is easier to validate that the HTML code is valid.

aledeg committed:
Add default redirect when authenticating (#4778)

math-GH committed:
Fix: swage menu (#4783)

* fix

* fix scroll bar

batchennings committed:
Mapco text display improvements (#4711)

* Main font color from blue to regular dark

* newline added at the end of rtl css file

* Text display improvements (code, quotes) + minor UI fixes

* replaced 2 spaces with 1 tab bcz phpcs is whining about that

* make fix-all

* Mapco/base-theme - CSS cleaning

* make fix-all

* PR #4711 cleaning

* Exclude generated CSS from lint check

* Revert exclusion of generated CSS files

Co-authored-by: patjennings <patjennings@MacBook-Pro-de-patjennings.local>
Co-authored-by: Alexandre Alapetite <alexandre@alapetite.fr>
Co-authored-by: patjennings <patjennings@macbook-pro-de-patjennings.home>
Co-authored-by: maTh <math-home@web.de>

math-GH committed:
icons (#4786)

math-GH committed:
fixes jumping lines (#4782)

* fixes

* fix template

* fix frss.css

* fix Nord theme

math-GH committed:
Fix: origine compact theme: line height (#4787)

* fix

* Origine

* Origine compact

math-GH committed:
fix: unread counter (#4788)

* fix

* fix

* fix

math-GH committed:
Improved: implement CSS variable --frss-padding-flux-items (#4789)

* frss.css

* mobile view: a bit compacter cols

math-GH committed:
fixed: box layout for global view (#4791)

math-GH committed:
improved: Pafat theme aside bar (#4792)

math-GH committed:
fix: .icon height (#4793)

math-GH committed:
fix swage (#4799)

math-GH committed:
fix: base theme: line-height (#4798)

Pyr0x1 committed:
Translated custom date/time labels in sub.php (#4803)

math-GH committed:
fix: Adark theme: line-height (#4797)

* fix

* fix

Pyr0x1 committed:
Added Pyr0x1 in CREDITS.md (#4805)

math-GH committed:
fix (#4806)

math-GH committed:
fix: Pafat theme: line-height (#4796)

* fix

math-GH committed:
fix: dyn. OPML icon (#4810)

* fix

Alkarex committed:
Exclude .git folder (#4824)

Alkarex committed:
Fix path_entries encoding (#4823)

* Fix path_entries encoding
#fix https://github.com/FreshRSS/FreshRSS/issues/4815

* Fix preview

Alkarex committed:
Avoid exception in searchById (#4822)

https://github.com/FreshRSS/FreshRSS/issues/4820
Not a root-cause fix though (cannot reproduce so far)

math-GH committed:
Improved: Swage theme with css vars (#4828)

* wip

* wip

* wip

* finished variables rewriting

* scrollbar - use default variables from frss.css

* fixes

* all colors are now in variables

* renaming variables

* reordering

* clean up

* add sourcemap link

* Update swage.scss

math-GH committed:
Improved: Swage theme, line-height without unit (#4829)

Alkarex committed:
Fix remember opened categories (#4827)

#fix https://github.com/FreshRSS/FreshRSS/issues/4826
Was crashing if a category did not exist anymore

math-GH committed:
fix: Pafat theme: aside navigation bar: icons (#4811)

* fix

* Warning sign style

Co-authored-by: Alexandre Alapetite <alexandre@alapetite.fr>

gingerbeardman committed:
Update README.md for ReadKit (#4710)

* Update README.md

I use ReadKit (free version) on iOS and macOS. 

Pro purchase is required to manage feeds, which I don't agree with given that we're using FreshRSS. Managing local feeds I can understand that could be value add.

* Tested + details

* Re-order based on capabilities

Co-authored-by: Alexandre Alapetite <alexandre@alapetite.fr>

math-GH committed:
Update update.phtml (#4830)

Alkarex committed:
Changelog 1.20.1

Alkarex committed:
More credits

math-GH committed:
Merge branch 'math-GH-wrap-titles' into wrap-titles

math-GH commented:
I lost control over the merging with main branch (about 100 commits and a lot of conflicts).

I created a new PR: #4857
